src/main/java/org/hibernate/infra/bot/EditPullRequestBodyAddIssueLinks.java

```diff
@@ -108,25 +108,20 @@ else if ( issueKeys.size() > repositoryConfig.jira.getIssueLinksLimit() ) {
 
 	private String constructLinksSection(Set<String> issueKeys, String body) {
 		final String lowerCaseBody = body.toLowerCase( Locale.ROOT );
-		final List<String> keys = new ArrayList<>( issueKeys.size() );
+		final List<String> links = new ArrayList<>( issueKeys.size() );
 		for ( String key : issueKeys ) {
 			// Add links for issue keys that are not already found in the original PR body
-			if ( !lowerCaseBody.contains( key.toLowerCase( Locale.ROOT ) ) ) {
-				keys.add( key );
+			String link = String.format( Locale.ROOT, LINK_TEMPLATE, key );
+			if ( !lowerCaseBody.contains( link.toLowerCase( Locale.ROOT ) ) ) {
```

Comment:
I know there are other ways to link to an issue, e.g.: https://hibernate.atlassian.net/jira/software/c/projects/HHH/issues/HHH-19277 or even https://hibernate.atlassian.net/jira/software/c/projects/HHH/list?selectedIssue=HHH-16897.

We *could* use a regex like `"https://hibernate.atlassian.net/[^\s]*" + key + "[^\s]*"` to match any link from Hibernate's Jira that contains correct issue key, but even then we cannot really be sure to match everything correctly. It's true that we usually prefer the "normal" link form, and normally external users just don't put anything in the PR's description, so 🤷

Reply:
yeah ... let's give it a try then, and if we end up having nondefault links used often, we can switch to a regex 🤞🏻 
thanks for having a look 😃 👍🏻